### DISCUSSION TITLE
Is a real two-way sync possible?

### DISCUSSION THREAD

**Discussion by janscholtes** [grouparoo/grouparoo#1519] [Q&A]
Thinking of Zendesk for example, we do not only what push data to zendesk, we also enrich our customer data through zendesk and want to push that information back to some database where it may gets pushed to other systems like pipedrive. When Zendesk is a "source" then, is this process possible in the current architecture or is there a need for a leading system where all data maintenance takes place? Can we achieve this by creating a own "zendesk source app" or is grouparoo not designed for this usecase?

**Comment by bleonard** [comment]
Hey @janscholtes, pulling data in from Zendesk is absolutely possible. Most of the current sources are databases, but we also support anyhting you can do with APIs. For example, we have a Mailchimp source to pull in people that sign up on their mailing list.

What are you looking to pull in?

**Reply by janscholtes** [reply]
Hi @bleonard, thank you for your fast reply. We want to sync our customer data between Zendesk, Pipedrive and Aircall.io. There are a few processes which we can't achieve with tools like piesync since we need to put some more logic in the syncing and because of some limitations regarding the fields which can be synced.

**Reply by bleonard** [reply]
What would you want to bring in from Zendesk?

Grouparoo would be well-suited to run a schedule that checks this [incremental API](https://developer.zendesk.com/rest_api/docs/support/incremental_export#incremental-user-export) and pull in information about the user. This would be all the things in that JSON (name, email, external_id, tags, custom fields, etc).

I'm a bit sad to see you can't get all their [related](https://developer.zendesk.com/rest_api/docs/support/users#show-user-related-information) information which includes a few ticket metrics without another call. There is something called [sideloading](https://developer.zendesk.com/rest_api/docs/support/side_loading#supported-endpoints) which pulls in the open_ticket_count, though.

**Reply by janscholtes** [reply]
We want to bring in all Information gathered there, for example we use some custom fields on the organization entity which may get populated during a support case. Is it from a today's perspective possible to do this with your pre-build zendesk app? In my understanding zendesk then also needs to be a source, so we need to develop an app / extend your app?

In your product concept the profile represents an individual person. How could an organization be represented? A profile property seems not the right place when thinking of a organization as an own entity with own information. A lot of crm systems follow this approach and there are for example no default fields for a telephone number of an organization in pipedrive (in hubspot there is one). For a lot of our daily usecases we need to focus on the organization entity and the individual profile of a person should us enable to get to the organizational entity/informations (Ticket counts, open deals, main telephone number, etc.), for example when we can match a callers number to a organization.

**Reply by bleonard** [reply]
Zendesk would definitely have to be a source to bring in data. There would have be a [source](https://www.grouparoo.com/docs/development/sources) plugin that would likely pull data in on a schedule.

We have the notion of [account/organizations/companies](https://github.com/grouparoo/grouparoo/issues/653) on our [roadmap](https://www.grouparoo.com/roadmap). The way that Grouparoo users are handling this now is using the `organizationId` (or other internal id) to duplicate the company scope data on the N profiles. Of course, this is less than ideal which is why we plan to update it, but it works.

There is also the downstream effect to consider. Zendesk does not have a "company" object that I'm aware of unless we are talking about a Custom Object. We have a User pipedrive integration, but I don't see the same concept of updating of company level info in their UI or [API](https://developers.pipedrive.com/docs/api/v1/#!/OrganizationFields/deleteOrganizationField). The point is, it's possible you want them on the individual users anyway (their individual activity and the company activity).

**Reply by bleonard** [reply]
Basically, it seems likely it will work. Let's chat about how to make it happen. Drop me a line: brian <at> grouparoo.com or get in our Slack: https://www.grouparoo.com/chat

**Reply by janscholtes** [reply]
> Zendesk does not have a "company" object that I'm aware of unless we are talking about a Custom Object. 

Zendesk has organizations to "segment customers into organizations", the entity can be found [here](https://developer.zendesk.com/rest_api/docs/support/organizations). The entity is especially relevant for tools like yours because it can only be added by admins or via api. There is a field `domain_names` which is used by zendesk to match incoming tickets from new contacts to the organization. For example you generate a lead and you know there domain name you can add the organization in zendesk (maybe coming from pipedrive) with this information. New contacts generated by a first-time contact creating a new ticket then gets automatically associated with this organization and this information then can also be shared through the source/destination mechanism (thats one process we want to achieve).

The pipedrive API reference (i like there API in most ways) seems not to be in full detail regarding the default fields they have on the organization entity, there are only a few:

![image](https://user-images.githubusercontent.com/34287604/112603535-6f6ffe00-8e15-11eb-834f-d5a8ab79f136.png)


Thank you for your detailed information, see you on Slack.

**Reply by bleonard** [reply]
I feel like we answered "Is two-way sync possible" here, but we haven't made the transition to chatting more on Slack or otherwise, so I didn't want to close it. feel free to email me (brian@grouparoo.com) or [Slack](https://www.grouparoo.com/chat).
